frame/composable-traits/src/dex.rs

```diff
@@ -152,6 +152,10 @@ pub struct ConstantProductPoolInfo<AccountId, AssetId> {
 	pub fee: Permill,
 	/// Amount of the fee goes to owner of the pool
 	pub owner_fee: Permill,
+	/// The weight of the base asset. Must hold `1 = base_weight + quote_weight`
+	pub base_weight: Permill,
```

Comment:
Wouldn't it be more forward compatible if we define this as an array of weights? Given that we want to support multi asset eventually? Obviously we can reject the case with more than 2 for now

Reply:
If we do so we should do it for add/remove liquidity as well and integrate a vec everywhere

Reply:
Note that the refactor will make a storage migration more difficult; I'd already refactor this to a Vec<Permill>, and check that len == 2; that way we can remove the check to allow multiweighted pools